### DISCUSSION TITLE
P20 Lite unlock

### DISCUSSION THREAD

**Discussion by kiemlicz** [kitsuned/PotatoNV#138] [Q&A] [answered]
Hello,
I have a Huawei P20 Lite which is stuck in a bootloop after some firmware upgrade.
I want just to recover the data
I cannot `fastboot boot twrp.img` due to errors
```
command not allowed
```
I guess I need to unlock the bootloader?
Does the unlocking the bootloader cause data loss?

EDIT: I've seend: https://github.com/mashed-potatoes/PotatoNV/discussions/24 but tbh I want to double check

**Comment by tmhakim** [comment, marked as answer]
Hello kiemlicz!

I guess I need to unlock the bootloader?

Yes you have to unlock the bootloader to flash TWRP on to your device

PS: you can't just type fastboot boot twrp.img it does not in huawei devices due to FBLOCK (Fastboot Lock). you have to type fastboot flash recovery_ramdisk twrp.img to flash TWRP

 Does the unlocking the bootloader cause data loss?

Yes it happens on all devices so it is recommended to have a backup or else your data is erased when bootloader is unlocked
